runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/testing/TestDataflowRunner.java

```diff
@@ -118,14 +120,16 @@ DataflowPipelineJob run(Pipeline pipeline, DataflowRunner runner) {
             try {
               for (;;) {
                 Optional<Boolean> result = checkForSuccess(job);
-                if (result.isPresent()) {
+                if (result.isPresent() && (!result.get() || atMaxWatermark(job))) {
```

Comment:
`checkForSuccess()` will return `Optional.absent()` if no metric present for job. In this case, I think `atMaxWatermark(job)` can be skipped. 

Also it's passible that the streaming pipeline doesn't use `PAssert`. So `checkForSuccess()` will return true before job is finished. That's why I want to return when only check `result.isPresent()`